invenio_assets/webpack.py

```diff
@@ -12,18 +12,50 @@
 from collections import OrderedDict
 
 from flask import current_app, request
-from flask_webpackext import WebpackBundle, WebpackBundleProject
+from flask_webpackext import WebpackBundle
+from flask_webpackext import WebpackBundleProject as WebpackBundleProjectBase
 from flask_webpackext.manifest import JinjaManifest, JinjaManifestLoader
+from invenio_base.utils import obj_or_import_string
 from markupsafe import Markup
 from pywebpack import ManifestEntry, UnsupportedExtensionError, bundles_from_entry_point
+from werkzeug.local import LocalProxy
 
-project = WebpackBundleProject(
+
+class WebpackBundleProject(WebpackBundleProjectBase):
+    """Flask webpack bundle project."""
+
+    def __init__(self, import_name, base_package_json="package.json", **kwargs):
+        """Constructor."""
+        super().__init__(import_name, **kwargs)
+        self._package_json_source_path = base_package_json
+
+    @property
+    def npmpkg(self):
```

Comment:
the super classes simply return `NPMPackage(self.path)` with no possibility to switch the package class.
introducing the new `ASSETS_NPM_PKG_CLS` (name suggestions welcome) config with this dynamic lookup mechanism was the most minimally invasive mechanism i could think of.

perhaps this could be lifted up into other packages, but that would of course require more PRs and releases.

Reply:
We lift this property to Flask-WebpackExt, to be controlled `WEBPACKEXT_NPM_PKG_CLS`.